src/interface/params.hpp

```diff
@@ -94,31 +90,11 @@ class Params {
   };
 
   template <typename T>
-  void typeCheck(const std::string key, bool die) {
+  void typeCheck(const std::string key) {
     // check on return type
-    // TODO(JMM): This is really clunky. Should we remove the conditional
-    // and put it in the require statement?
-    if (myTypes_[key].compare(std::string(typeid(T).name()))) {
-      std::string message = "WRONG TYPE FOR KEY '" + key + "'";
-      PARTHENON_REQUIRE(!die, message.c_str());
-    }
-  }
-
-  void keyCheck(const std::string key, bool die) {
-#ifndef NDEBUG
-    if (!hasKey(key)) {
-      // key alread exists, replace
-      std::cout << std::endl;
-      std::cout << "----------------------------------------------" << std::endl;
-      std::cout << "    WARNING: key '" << key << "' not found." << std::endl;
-      std::cout << "             Swift death will follow..." << std::endl;
-      std::cout << "----------------------------------------------" << std::endl;
-      std::cout << std::endl;
-
-      std::string message = "Key " + key + " doesn't exist";
-      PARTHENON_DEBUG_REQUIRE(!die, message.c_str());
-    }
-#endif // NDEBUG
+    bool badtype = myTypes_[key].compare(std::string(typeid(T).name()));
+    std::string message = "WRONG TYPE FOR KEY '" + key + "'";
+    PARTHENON_REQUIRE(!badtype, message.c_str());
```

Comment:
Since you've reduced this to a single line, does it make sense to eliminate the typeCheck() procedure and instead absorb it above where it is called (since that is the only instance?)

Reply:
Yes I'll fix that.